pkg/apis/resource/types.go

```diff
@@ -128,13 +128,35 @@ type ResourceSliceSpec struct {
 	// +oneOf=NodeSelection
 	AllNodes bool
 
-	// Devices lists some or all of the devices in this pool.
+	// Devices lists some or all of the devices in this pool. Devices
+	// listed here do not inherit shared attributes from the slice.
 	//
 	// Must not have more than 128 entries.
 	//
 	// +optional
 	// +listType=atomic
 	Devices []Device
+
+	// SharedCapacity defines the set of shared capacity consumable by
+	// partitionable devices in this ResourceSlice.
+	//
+	// Must not have more than 128 entries. Names must be C identifiers
+	// that are unique inside the ResourceSlice.
+	//
+	// +listType=atomic
+	// +optional
+	SharedCapacity map[string]resource.Quantity
+
+	// Attributes contains common device attributes that are the same for
+	// all partitionable devices in the pool, unless a device specifically
+	// over writes it by defining an attribute of the same name.
+	Attributes map[QualifiedName]DeviceAttribute
+
+	// Devices lists some or all of the partitionable devices in this pool.
+	// Partitionable devices may inherit shared attributes from the
+	// ResourceSlice and they may (but don't have to) consume shared
+	// capacity.
+	PartitionableDevices []PartitionableDevice
```

Comment:
+optional, +listType=atomic

Reply:
I am very much against embedding partitions like this. I haven't had time to spec out my proposal (which never made it into John's branch of proposals in the prototype). But, in essence, my proposal retains a flat list of devices with the same fields it has today, with the additional ability to reference a separate object that defines it's "shape". This "shape" object then allows for things attribute inheritance, etc.  but a direct device does not.